### PR TITLE
Libreoffice refresh after insert citation

### PR DESCRIPTION
**Pull request by mohandhandapani**
Added new funtion(Remove_Bookmark) to remove exiting mendeley bookmarks after export to MS-Word.

MD-22156

### CODE REVIEW

File: src/mendeleyMain.vb [hunk 1]
```diff
@@ -692,6 +692,35 @@ EndOfSub:
...
 End Sub
 
+'Remove Bookmarks after export to MS-WORD
+Sub Remove_Bookmark()
+On Error GoTo ErrorHandler
+     Dim mBookmarks,rbTemp
+     Dim i As Long, j As Long, m as Long
+     Dim args1(0) as new com.sun.star.beans.PropertyValue
+
+rem define variables
```
Comment (cpina) [inline src/mendeleyMain.vb]: Please delete commented out code

File: src/mendeleyMain.vb [hunk 2]
```diff
@@ -692,6 +692,35 @@ EndOfSub:
     uiDisabled = False
 End Sub
 
+'Remove Bookmarks after export to MS-WORD
+Sub Remove_Bookmark()
+On Error GoTo ErrorHandler
```
Comment (cpina) [inline src/mendeleyMain.vb]: Have a look how the ErrorHandler is used in the plugin - I think that we should use the same approach (if not debug: executes reportError, if debug then no error handler which helps us fixing things instead of ignoring the errors)

File: src/mendeleyMain.vb [hunk 3]
```diff
@@ -692,6 +692,35 @@ EndOfSub:
...
+	If UBound(mBookmarks) = 0 then
+		Exit Sub
+	End If
+    For j = 0 To UBound(mBookmarks)
+        If (Left(mBookmarks(j), 9) = "Mendeley_" Or InStr(mBookmarks(j), "CSL_CITATION") > 0 Or InStr(mBookmarks(j), "CSL_BIBLIOGRAPHY") > 0) Then
+             args1(0).Name = "Bookmark"
+             args1(0).Value = mBookmarks(j)
+             dispatcher.executeDispatch(document, ".uno:DeleteBookmark", "", 0, args1())
+         End if
+    i = i + 1
```
Comment (cpina) [inline src/mendeleyMain.vb]: Why is this line needed? "j" is the index for the For and fetching the bookmarks...

File: src/mendeleyMain.vb [hunk 4]
```diff
@@ -692,6 +692,35 @@ EndOfSub:
     uiDisabled = False
 End Sub
 
+'Remove Bookmarks after export to MS-WORD
+Sub Remove_Bookmark()
+On Error GoTo ErrorHandler
+     Dim mBookmarks,rbTemp
+     Dim i As Long, j As Long, m as Long
```
Comment (cpina) [inline src/mendeleyMain.vb]: I don't think that 'm' is used - then better delete it

**Comment by mohandhandapani** [comment]
Removed unnecessary code,commented line and included error handler condition

File: src/mendeleyMain.vb [hunk 5]
```diff
@@ -692,6 +692,33 @@ EndOfSub:
     uiDisabled = False
 End Sub
 
+Sub Remove_Bookmark()
+On Error GoTo ErrorHandler
```
Comment (cpina) [inline src/mendeleyMain.vb]: Question: because Remove_Bookmark() is called from insertCitationButton() and this oen is already setting up the ErrorHandler: is it needed here? I don't know if the ErrorHandler is propagated into other functions or needs to be set again.

And if needs to be done please do like everywhere else:
    If Not DEBUG_MODE Then
        On Error GoTo ErrorHandler
    End If

So we have the same type of code.

File: src/mendeleyMain.vb [hunk 6]
```diff
@@ -692,6 +692,33 @@ EndOfSub:
...
+     Dim mBookmarks,rbTemp
+     Dim j As Long
+     Dim args1(0) as new com.sun.star.beans.PropertyValue
+     Dim document   as object
+     Dim dispatcher as object
+
+     document   = ThisComponent.CurrentController.Frame
+     dispatcher = createUnoService("com.sun.star.frame.DispatchHelper")
+     mBookmarks = thisComponent.Bookmarks.ElementNames
+    If UBound(mBookmarks) = 0 then
```
Comment (cpina) [inline src/mendeleyMain.vb]: I don't think that this If is needed - the below "For j = 0 To UBound(mBookmarks)" will not iterate if mBookmarks is empty.

**Comment by mohandhandapani** [comment]
For loop is to check/delete each mendeley bookmarks, If bookmark is empty it will not goto loop becuase already added condition for validate the empty state of Bookmark.

**Comment by mohandhandapani** [comment]
Removed addtional condition and error handler

**Comment by mohandhandapani** [comment]
Removed and function other area and  kept only call  "Remove_Bookmark" function after exportAsBookmarks.
